### PR TITLE
Framework: use social-logos package

### PR DESCRIPTION
**Pull request by gwwar**
Instead of manually copying from https://github.com/Automattic/social-logos this PR pairs with https://github.com/Automattic/social-logos/pull/42 to use it as an external dependency.

Note that this component assumes that `react` and `classnames` are available in the parent project.

```javascript
import SocialLogo from 'social-logos';
<SocialLogo icon="twitter" size={ 48 } />
```

Misc changes have snuck into the Calypso SocialLogo file over time, so keep an eye out for any breakages. For example, using service names directly eg `google_plus` and failing to transform the value into kebobCase `google-plus`

### Testing
- No functional changes
- Social Logos should appear correctly in http://calypso.localhost:3000/devdocs/design/social-logos
- On http://calypso.localhost:3000/sharing/:siteId:
- And in the reader? cc @blowery 

## Tasks
- After https://github.com/Automattic/social-logos/pull/42  lands update dep version to 1.0.0 instead of alpha

cc @aduth @obenland @lamosty @retrofox

### CODE REVIEW

File: client/components/social-logo/README.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
 Social Logo
 ========
 
-Component that renders a single socail-logo svg based on an `icon` prop. It takes a size property but defaults to 24px. For greater sharpness, the icons should only be shown at either 18, 24, 36 or 48px. There's a gallery with all the available icons in devdocs/design.
```
Comment (lamosty) [inline client/components/social-logo/README.md]: Could we maybe add a link to https://github.com/Automattic/social-logos in the README? I wouldn't know we are speaking about that package if I didn't see this PR.

Maybe just adding a link to the text: "...renders a single [social-logo](link here) svg based...".

File: client/components/social-logo/README.md [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+Component that renders a single social-logo svg based on an `icon` prop. It takes a size property but defaults to 24px. For greater sharpness, the icons should only be shown at either 18, 24, 36 or 48px. There's a gallery with all the available icons in devdocs/design.
```
Comment (retrofox) [inline client/components/social-logo/README.md]: stupid comment but should we add `px` to each size: `18px, 24px, 36px or 48px

**Comment by lamosty** [comment]
Nice work!

LGTM :ship:.

File: client/components/social-logo/README.md (hunk 2 above)
Comment (gwwar) [inline client/components/social-logo/README.md]: Thanks I'll add this to the social-logo readme

File: client/components/social-logo/README.md (hunk 1 above)
Comment (gwwar) [inline client/components/social-logo/README.md]: Sounds good 👍  I'll try to follow up in a later PR to show examples in devdocs directly from the package vs here.

**Comment by gwwar** [comment]
Thanks for the reviews!

**Pull request merged**